src/main/webapp/src/app/recommendation-form/recommendation-form.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, FormControl} from '@angular/forms';
+
+@Component({
+  selector: 'app-recommendation-form',
+  templateUrl: './recommendation-form.component.html',
+  styleUrls: ['./recommendation-form.component.css']
+})
+export class RecommendationFormComponent implements OnInit {
+
+  // Build preference form
+  preferenceForm = this.fb.group({
+    cuisine: "",
+    price: "",
+    rating:"",
+    type: ""
+  });
+
+  constructor(private fb: FormBuilder) { }
```

Comment:
since this is never re-assigned it would be good to make it readonly, eg private readonly fb 
also it might be nice to user formBuilder instead of fb to make it more readable.

Reply:
done